swarm/src/behaviour.rs

```diff
@@ -630,7 +630,20 @@ pub enum NetworkBehaviourAction<
     /// # #[derive(Debug, PartialEq, Eq)]
     /// # struct PreciousMessage(String);
     /// ```
-    Dial { opts: DialOpts, handler: THandler },
+    Dial {
+        opts: DialOpts,
+        handler: THandler,
+    },
+
+    /// Instructs the `Swarm` to listen on the provided address
+    ListenOn {
+        address: Multiaddr,
+    },
```

Comment:
With what we figured out in the discussion around #2824, we should provide a mechanism on how to associate a listening request with the later on produced events.

To follow what we will likely end up doing in https://github.com/libp2p/rust-libp2p/pull/3254, I think we should pass the `ListenerId` into here already. That is a bit tricky with the current `Transport` interface though.

Reply:
Related: https://github.com/libp2p/rust-libp2p/pull/3290